js/site-search.js

```diff
@@ -1,41 +1,79 @@
 namespacer('bcpl');
 
-bcpl.siteSearch = (($) => {
+bcpl.siteSearch = (($, window, constants) => {
 	const siteSearchTabSelector = '.search-button';
 	const siteSearchInputSelector = '#site-search-input';
-	const siteSearchClearIconSelector = '.site-search-input-container .fa-times';
 	const siteSearchSearchIconSelector = '.site-search-input-container .fa-search';
+	const searchButtonCatalogSelector = '.search-button-catalog';
+	const searchButtonEventsSelector = '.search-button-events';
+	const searchButtonWebsiteSelector = '.search-button-website';
+	const searchAction = {};
 
 	const onSearchTabClick = (clickEvent) => {
-		const $searchBtn = $(clickEvent.target);
+		const $searchBtn = $(clickEvent.currentTarget);
 		$searchBtn
 			.siblings().removeClass('active').end()
 			.addClass('active');
 	};
 
-	const onSearchClearBtnClick = () => {
-		$(siteSearchInputSelector)
-			.val('')
-			.trigger('keyup')
-			.focus();
+	const onSearchCatalogClick = () => {
+		searchAction.search = () => { searchCatalog(window); };
 	};
 
-	const onSearchInputKeyup = (keyupEvent) => {
-		const $searchInput = $(keyupEvent.target);
-		const doesSearchHaveValue = $searchInput.val();
-		let $elmToHide = $(siteSearchSearchIconSelector);
-		let $elmToShow = $(siteSearchClearIconSelector);
+	const onSearchIconClick = () => {
+		const searchTerms = getSearchTerms();
 
-		if (!doesSearchHaveValue) {
-			$elmToHide = $(siteSearchClearIconSelector);
-			$elmToShow = $(siteSearchSearchIconSelector);
+		if (searchAction && searchAction.search && searchTerms.length) {
```

Comment:
i dont think this would ever be an issue, but is it worth checking to see if .search is a function?